packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -0,0 +1,50 @@
+import * as cdk from '@aws-cdk/core';
+import { TagUpdateConstraintOptions } from '../constraints';
+import { IPortfolio } from '../portfolio';
+import { IProduct } from '../product';
+import { CfnPortfolioProductAssociation, CfnResourceUpdateConstraint } from '../servicecatalog.generated';
+import { hashValues } from './util';
+import { InputValidator } from './validation';
+
+export class AssociationManager {
+  public static associateProductWithPortfolio(
+    portfolio: IPortfolio, product: IProduct,
+  ): { associationKey: string, cfnPortfolioProductAssociation: CfnPortfolioProductAssociation } {
+    const associationKey = hashValues(portfolio.node.addr, product.node.addr, portfolio.stack.node.addr);
+    const constructId = `PortfolioProductAssociation${associationKey}`;
+    const associationNode = portfolio.node.tryFindChild(constructId);
```

Comment:
The name `node` is weird - this is not a "node", this is actually the Construct. So I would call this just `existingAssociation`, or something like that.

Reply:
donne